<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="author" content="Tade Souaiaia" />
      <link rel="shortcut icon" href="../../img/favicon.ico" />
    <title>Day1 (AM) - prsWorkshop</title>
    <link rel="stylesheet" href="../../css/theme.css" />
    <link rel="stylesheet" href="../../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
        <link href="../../css/extra.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Day1 (AM)";
        var mkdocs_page_input_path = "practicals/workshop_practical_paul1.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="../.." class="icon icon-home"> prsWorkshop
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../..">Home</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">Pre-Workshop</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../prep_list/">Checklist</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../prep_os/">Operating System</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../prep_terminal/">Terminal Setup</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../prep_software/">Software Installs</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../prep_testing/">Testing</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Pre-Workshop Tutorials</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../tut_intro/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../tut_bash/">Bash Tutorial</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../tut_R/">R Tutorial</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../tut_python/">Python Tutorial</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../tut_plink/">Plink Tutorial</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Program</span></p>
              <ul class="current">
                  <li class="toctree-l1 current"><a class="reference internal current" href="./">Day1 (AM)</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#11-key-learning-outcomes">1.1 Key Learning Outcomes</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#12-resources-you-will-be-using">1.2 Resources you will be using</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#13-prs-definition">1.3 PRS Definition</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#14-understanding-gwas-summary-statistics">1.4 Understanding GWAS Summary Statistics</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#15-matching-the-base-and-target-data-sets">1.5 Matching the Base and Target Data sets</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#16-linkage-disequilibrium-in-prs-analyses">1.6 Linkage Disequilibrium in PRS Analyses</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#161-performing-clumping">1.6.1 Performing Clumping</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#17-running-prsice-the-ct-method">1.7 Running PRSice: the "C+T" method</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#171-height-prs-using-gw-significant-snps-only">1.7.1 Height PRS using GW-significant SNPs only</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#172-height-prs-across-multiple-p-value-thresholds">1.7.2 Height PRS across multiple P-value thresholds</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#173-high-resolution-scoring">1.7.3 High-Resolution Scoring</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#18-accounting-for-covariates">1.8 Accounting for Covariates</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#19-case-control-studies">1.9 Case-Control Studies</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#110-generating-prs-quantile-plots">1.10 Generating PRS Quantile Plots</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#111-cross-trait-analysis">1.11 Cross-Trait Analysis</a>
    </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../workshop_practical_tade/">Day1 (PM)</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../workshop_practical_clive/">Day2 (AM)</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../workshop_practical_paul2/">Day2 (PM)</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Misc</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../misc_linux/">Linux For Windows</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../misc_plink_problem/">Mac Security</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../..">prsWorkshop</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../.." class="icon icon-home" aria-label="Docs"></a></li>
          <li class="breadcrumb-item">Program</li>
      <li class="breadcrumb-item active">Day1 (AM)</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/tadesouaiaia/prsWorkshop-website/edit/master/docs/practicals/workshop_practical_paul1.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="introduction-to-polygenic-scores-analyses">Introduction to Polygenic Scores Analyses</h1>
<h2 id="11-key-learning-outcomes">1.1 Key Learning Outcomes</h2>
<p>After completing this practical, you should be able to:</p>
<ol>
<li>Perform basic Polygenic Risk Score (PRS) analyses using PRSice:
    (<a href="https://doi:10.1093/bioinformatics/btu848">Euesden, Lewis &amp; O'Reilly 2015</a>;<a href="https://doi:10.1093/gigascience/giz082">Choi &amp; O'Reilly 2019</a>)</li>
<li>Interpret results generated from PRS analyses</li>
<li>Customise visualisation of results</li>
</ol>
<h2 id="12-resources-you-will-be-using">1.2 Resources you will be using</h2>
<p>To perform PRS analyses, summary statistics from Genome-Wide Association Studies (GWAS) are required. In this workshop, the following summary statistics are used:</p>
<table>
<thead>
<tr>
<th style="text-align: center;"><strong>Phenotype</strong></th>
<th style="text-align: center;"><strong>Provider</strong></th>
<th style="text-align: center;"><strong>Description</strong></th>
<th style="text-align: center;"><strong>Download Link</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: center;">Height</td>
<td style="text-align: center;"><a href="https://portals.broadinstitute.org/collaboration/giant/index.php/GIANTconsortium_data_files">GIANT Consortium</a></td>
<td style="text-align: center;">GWAS of height on 253,288 individuals</td>
<td style="text-align: center;"><a href="https://portals.broadinstitute.org/collaboration/giant/images/0/01/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz">Link</a></td>
</tr>
<tr>
<td style="text-align: center;">Coronary artery disease (CAD)</td>
<td style="text-align: center;"><a href="http://www.cardiogramplusc4d.org/">CARDIoGRAMplusC4D Consortium</a></td>
<td style="text-align: center;">GWAS on 60,801 CAD cases and 123,504 controls</td>
<td style="text-align: center;"><a href="http://www.cardiogramplusc4d.org/media/cardiogramplusc4d-consortium/data-downloads/cad.additive.Oct2015.pub.zip">Link</a></td>
</tr>
</tbody>
</table>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>All target phenotype data in this worshop are <strong>simulated</strong>. They have no specific biological meaning and are for demonstration purposes only. </p>
</div>
<h2 id="13-prs-definition">1.3 PRS Definition</h2>
<p>A Polygenic Risk Score (PRS) is an estimate of an individual’s genetic propensity to a phenotype, calculated as a sum of their genome-wide genotypes, weighted by corresponding genotype effect sizes obtained from GWAS summary statistics. 
In the next section we will consider what the effect size means and how it is used in computing PRS. </p>
<h2 id="14-understanding-gwas-summary-statistics">1.4 Understanding GWAS Summary Statistics</h2>
<p>When GWAS are performed on a quantitative trait, the effect size is
typically given as a beta coefficient (<span class="arithmatex">\(\beta\)</span>) from a linear regression
 with Single Nucleotide Polymorphism (SNP) genotypes as predictor of
 phenotype. The <span class="arithmatex">\(\beta\)</span> coefficient estimates the increase in the phenotype
for each copy of the <em>effect allele</em>. For example, if the effect
 allele of a SNP is <strong>G</strong> and the non-effect allele is <strong>A</strong> , then the
 genotypes <strong>AA</strong>, <strong>AG</strong> and <strong>GG</strong> will be coded as 0, 1 and 2
 respectively. In this scenario, the <span class="arithmatex">\(\beta\)</span> coefficient reflects how much
 the phenotype changes for each <strong>G</strong> allele present (NB. The <span class="arithmatex">\(\beta\)</span> can be
 positive or negative - so the 'effect allele' is simply the allele
 that was coded in the regression, not necessarily the allele with a
 positive effect).</p>
<p>When a GWAS is performed on a binary trait (e.g. case-control study),
 the effect size is usually reported as an Odd Ratios (OR). Using the
 same example, if the OR from the GWAS is 2 with respect to the <strong>G</strong>
 allele, then the OR of <strong>AG</strong> relative to <strong>AA</strong> is 2, and the OR of
 <strong>GG</strong> relative to <strong>AA</strong> is 4. So an individual with the <strong>GG</strong>
 genotype are estimated* to be 4 times more likely to be a case than
 someone with the <strong>AA</strong> genotype (*an Odds Ratio is itself an
 estimate of a Risk Ratio, which cannot be calculated from a
 case/control study)</p>
<hr />
<div class="admonition information">
<p class="admonition-title">Information</p>
<p>The relationship between the <span class="arithmatex">\(\beta\)</span> coefficient from the logistic regression and the OR is: 
 $$
 OR = e ^ \beta
 $$
 and 
 $$
 log_{e}(OR) = \beta
 $$</p>
<p>While GWAS sometimes convert from the <span class="arithmatex">\(\beta\)</span> to the OR when reporting results, most PRS software convert OR back to <span class="arithmatex">\(\beta\)</span>'s (<span class="arithmatex">\(log_e(OR)\)</span>) to allow simple addition of <span class="arithmatex">\(log_e(OR)\)</span>'s.</p>
</div>
<div class="admonition questions">
<p class="admonition-title">Questions</p>
<p>Column names are not standardised across reported GWAS results, thus it is important to check which column is the effect (coded) allele and which is the non-effect allele. For example, in the height GWAS conducted by the GIANT consortium, the effect allele is in the column Allele1, while Allele2 represents the non-effect allele.</p>
<p>Let us open the Height GWAS file (<strong>day1a/Base_Data/GIANT_Height.txt</strong>) and inspect the SNPs at the top of the file. If we only consider SNPs <em>rs4747841</em> and <em>rs878177</em>, what will the ‘PRS’ of an individual with genotypes <strong>AA</strong> and <strong>TC</strong>, respectively, be? And what about for an individual with <strong>AG</strong> and <strong>CC</strong>, respectively? (Careful these are not easy to get correct! This shows how careful PRS algorithms/code need to be).</p>
<p>What do these PRS values mean in terms of the height of those individuals?</p>
</div>
<hr />
<h2 id="15-matching-the-base-and-target-data-sets">1.5 Matching the Base and Target Data sets</h2>
<p>The first step in PRS calculation is to ensure consistency between the GWAS summary statistic file (<em>base data</em>) and the target genotype file (<em>target data</em>). Since the base and target data are generated independently, they often relate to different SNPs and so the first job is to identify the overlapping SNPs across the two data sets and remove non-overlapping SNPs (this is usually done for you by PRS software). If the overlap is low then it would be a good idea to perform imputation on your target data to increase the number of SNPs that overlap between the data sets.</p>
<p>The next, more tricky issue, is that the genotype encoding between the data sets may differ. For example, while the effect allele of a SNP is <strong>T</strong> in the base data, the effect allele in the target might be <strong>G</strong> instead. When this occurs, <em>allele flipping</em> should be performed,where the genotype encoding in the target data is reversed so that <strong>TT</strong>, <strong>TG</strong> and <strong>GG</strong> are coded as 2, 1 and 0. Again, this is usually performed automatically by PRS software.</p>
<hr />
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>For SNPs that have complementary alleles, e.g. <strong>A/T</strong>, <strong>G/C</strong>, we cannot be certain that the alleles referred to in the target data correspond to those of the base data or whether they are the 'other way around' due to being on the other DNA strand (unless the same genotyping chip was used for all data). These SNPs are known as <strong><em>ambiguous SNPs</em></strong>, and while allele frequency information can be used to match the alleles, we remove ambiguous SNPs in PRSice to avoid the possibility of introducting unknown bias.</p>
</div>
<h2 id="16-linkage-disequilibrium-in-prs-analyses">1.6 Linkage Disequilibrium in PRS Analyses</h2>
<p>GWAS are typically performed one-snp-at-a-time, which, combined with the strong correlation structure across the genome (Linkage Disequilibrium (LD)), makes identifying the independent causal genetic effects challenging. There are two main options to address this challenge when calculating PRS: 1. SNPs are <em>clumped</em> so that the retained SNPs are largely independent of each other, allowing their effects to be summed, assuming additive effects, 2. all SNPs are included and the LD between them is accounted for. </p>
<p>While option 2 is statistically appealing, option 1 has been most adopted in PRS studies so far, most likely due to its simplicity and the similarity of results of methods using the different options to date (Pain et al 2021). In this workshop we will consider option 1 (the "clumping+thresholding", or C+T, method), implemented in PRSice (Choi &amp; O'Reilly 2019), but if you are interested in how LD can be incorporated as a parameter in PRS calculation then see the LDpred (Prive et al 2020), PRS-CS (Ge et al 2019) and SBayesRC (Zheng et al 2024) papers. </p>
<hr />
<h3 id="161-performing-clumping">1.6.1 Performing Clumping</h3>
<p>Clumping is the procedure where a SNP data set is `thinned' by removing SNPs across the genome that are correlated (in high LD) with a nearby SNP that has a smaller association <span class="arithmatex">\(P\)</span>-value. </p>
<p>SNPs are first sorted (i.e. ranked) by their <span class="arithmatex">\(P\)</span>-values. Then, starting from the most significant SNP (denoted as the <em>index SNP</em>), any SNPs in high LD (eg. <span class="arithmatex">\(r^2\)</span> &gt; 0.1) with the index SNP are removed. To reduce computational burden, only SNPs that are within e.g. 250 kb of the <em>index SNP</em> are <em>clumped</em>. This process is continued until no <em>index SNPs</em> remain. </p>
<p>Use the command below to perform clumping of the Height GWAS data using PLINK (Chang et al 2015). 
First, you will have to navigate to the <code>website_practical_downloads/day1a</code> folder using the terminal. </p>
<p>Next type the following command (NB. See warning below):</p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>./Software/plink_mac \
     --bfile Target_Data/TAR \
     --clump Base_Data/GIANT_Height.txt \
     --clump-p1 1 \
     --clump-snp-field MarkerName \
     --clump-field p \
     --clump-kb 250 \
     --clump-r2 0.1 \
     --out Results/Height
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can copy &amp; paste code from this document directly to the terminal.</p>
</div>
<p>The command above performs clumping on the height GWAS using LD calculated based on the <strong>TAR</strong> genotype file. SNPs that have <span class="arithmatex">\(r^2&gt;0.1\)</span> within a 250 kb window of the index SNPs are removed. This will generate the <strong>Height.clumped</strong> file, which contains the SNPs retained after clumping. </p>
<div class="admonition questions">
<p class="admonition-title">Questions</p>
<p>How many SNPs were in the GIANT_Height.txt file before clumping? </p>
<p>How many SNPs remain after clumping?</p>
<p>If we change the <span class="arithmatex">\(r^2\)</span> threshold to 0.2, how many SNPs remain? Why are there now more SNPs remaining?</p>
<p>Why is clumping performed for calculation of PRS? (in the standard approach)</p>
</div>
<h2 id="17-running-prsice-the-ct-method">1.7 Running PRSice: the "C+T" method</h2>
<p>Now that we know how to perform the <em>clumping</em> part of the "clumping+thresholding" (C+T) method, in this section we will learn about the <em>thresholding</em> part, and we will learn how to run PRSice, which performs both parts (C+T) simultaneously. </p>
<p>Deciding which SNPs to include is one of the key challenges in the calculation of PRS. 
A simple and popular approach is to include SNPs according to their GWAS association <span class="arithmatex">\(P\)</span>-value. 
For example, we may choose to include only the genome-wide significant SNPs from the GWAS because those are the SNPs with significant evidence for association. 
In the next subsection you will compute PRS from GW-significant SNPs only, and then in the subsequent subsection you will generate multiple PRSs using different <span class="arithmatex">\(P\)</span>-value thresholds. </p>
<h3 id="171-height-prs-using-gw-significant-snps-only">1.7.1 Height PRS using GW-significant SNPs only</h3>
<p>Use the commands below to run PRSice with GIANT Height GWAS as base data and the height phenotype target data. 
PRSice will calculate Height PRS in the target data and then perform a regression of the Height PRS against the target individual's true height values. 
From the <code>day1a/</code> directory, run the following command in the terminal:</p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>Rscript ./Software/PRSice.R \
     --prsice Software/PRSice_mac \
     --base  Base_Data/GIANT_Height.txt  \
     --target Target_Data/TAR \
     --snp MarkerName \
     --A1 Allele1 \
     --A2 Allele2 \
     --stat b \
     --beta \
     --pvalue p \
     --pheno Target_Data/TAR.height \
     --binary-target F \
     --bar-levels 5e-8 \
     --no-full \
     --fastscore \
     --out Results/Height.gws
</code></pre></div>
<p>This command takes the Height GWAS summary statistic file (<code>--base</code>), informs PRSice of the column name for the column containing the SNP ID (<code>--snp</code>), the effect allele (<code>--A1</code>), the non-effect allele (<code>--A2</code>), the effect size (<code>--stat</code>) and the <span class="arithmatex">\(P\)</span>-value (<code>--pvalue</code>).
We also inform PRSice that the effect size is a <span class="arithmatex">\(\beta\)</span> coefficient (<code>--beta</code>) instead of an OR.
The <code>--binary-target F</code> command informs PRSice that the target phenotype is a quantitative trait and thus linear regression should be performed. 
In addition, we ask PRSice not to perform high-resolution scoring over multiple thresholds (--fastscore), and to compute the PRS using only those SNPs with <span class="arithmatex">\(P\)</span>-value <span class="arithmatex">\(&lt; 5 \times 10^{-8}\)</span> (<code>--bar-levels</code> and <code>--no-full</code>).</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The default of PRSice is to perform clumping with an <span class="arithmatex">\(r^2\)</span> threshold of 0.1 and a window size of 250kb.</p>
<p>To see a full list of command line options available in PRSice type:</p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>./Software/PRSice_mac -h
</code></pre></div>
<p>Take some time to look through some of these user options and you should be able to see why the above command produced the desired output. </p>
</div>
<p>PRSice performs strand flipping and clumping automatically and generates the <strong>Height.gws.summary</strong> file, together with other output that we will look into later in the practical.
The summary file contains the following columns:</p>
<ol>
<li><strong>Phenotype</strong> - Name of Phenotype</li>
<li><strong>Set</strong> - Name of Gene Set. Default is <em>Base</em></li>
<li><strong>Threshold</strong> - Best P-value Threshold</li>
<li><strong>PRS.R2</strong> - Variance explained by the PRS</li>
<li><strong>Full.R2</strong> - Variance explained by the full model (including the covariates)</li>
<li><strong>Null.R2</strong> - Variance explained by the covariates (none provided here)</li>
<li><strong>Prevalence</strong> - The population disease prevalence as indicated by the user (not provided here due to testing continuous trait)</li>
<li><strong>Coefficient</strong> - The <span class="arithmatex">\(\beta\)</span> coefficient corresponding to the effect estimate of the best-fit PRS on the target trait in the regression. A one unit increase in the PRS increases the outcome by <span class="arithmatex">\(\beta\)</span></li>
<li><strong>Standard.Error</strong> - The standard error of the best-fit PRS <span class="arithmatex">\(\beta\)</span> coefficient (see above)</li>
<li><strong>P</strong> - The <span class="arithmatex">\(P\)</span>-value relating to testing the null hypothesis that the best-fit PRS <span class="arithmatex">\(\beta\)</span> coefficient is zero.</li>
<li><strong>Num_SNP</strong> - Number of SNPs included in the best-fit PRS</li>
<li><strong>Empirical-P</strong> - Only provided if permutation is performed. This is the empirical <span class="arithmatex">\(P\)</span>-value corresponding to the association test of the best-fit PRS - this controls for the over-fitting that occurs when multiple thresholds are tested.</li>
</ol>
<p>For now, we can ignore most columns and focus on the <strong>PRS.R2</strong> and the <strong>P</strong> column, which provide information on the model fit.</p>
<div class="admonition questions">
<p class="admonition-title">Questions</p>
<p>What is the <span class="arithmatex">\(R^2\)</span> for the PRS constructed using only genome-wide significant SNPs?</p>
<p>What is the <span class="arithmatex">\(P\)</span>-value for the association between the PRS and the outcome? Is this significant? (explain your answer). </p>
</div>
<hr />
<h3 id="172-height-prs-across-multiple-p-value-thresholds">1.7.2 Height PRS across multiple P-value thresholds</h3>
<p>A disadvantage of using only genome-wide significant SNPs is that there are likely to be many true signals among SNPs that did not reach genome-wide significance. 
However, since we do not know what <span class="arithmatex">\(P\)</span>-value threshold provides the "best" prediction for our particular data, then we can calculate the PRS at several <span class="arithmatex">\(P\)</span>-value thresholds and test their prediction accuracy to identify the "best" threshold (NB. See Dudbridge et al 2013 for theory on factors affecting the best threshold). 
This process is implemented in PRSice and can be performed automatically as follows:</p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>Rscript ./Software/PRSice.R \
     --prsice Software/PRSice_mac \
     --base  Base_Data/GIANT_Height.txt \
     --target Target_Data/TAR \
     --snp MarkerName \
     --A1 Allele1 \
     --A2 Allele2 \
     --stat b \
     --beta \
     --pvalue p \
     --pheno Target_Data/TAR.height \
     --binary-target F \
     --fastscore \
     --out Results/Height.fast
</code></pre></div>
<p>By removing the <code>--bar-levels</code> and <code>--no-full</code> command, we ask PRSice to perform PRS calculation with a number of predefined thresholds (0.001, 0.05, 0.1, 0.2, 0.3, 0.4, 0.5, 1). 
The <strong>.prsice</strong> file is very similar to the <strong>.summary</strong> file, the only difference is that <strong>.prsice</strong> file reports the results of model fits for <strong>all thresholds</strong> instead of the most predictive threshold. 
This allow us to observe the change in model fitting across different thresholds, visualized by the BARPLOT (Fig.1.1) generated by PRSice. Open up the bar plot image file in the Results/ folder and take a look at the bar plot. </p>
<p><img alt="BARPLOT generated by PRSice" src="../../images/PRSice_BARPLOT.png" /></p>
<div class="admonition questions">
<p class="admonition-title">Questions</p>
<p>Which is the most predictive threshold?</p>
<p>What is the <span class="arithmatex">\(R^2\)</span> of the most predictive threshold and how does it compare to PRS generated using only genome-wide significant SNPs?</p>
</div>
<h3 id="173-high-resolution-scoring">1.7.3 High-Resolution Scoring</h3>
<p>If we only calculate PRS at a small number of <span class="arithmatex">\(P\)</span>-value thresholds, then we might not observe the most predictive threshold. In PRSice, we can calculate and test PRS at a large number of thresholds using "high-resolution scoring" - which we can perform simply by removing the <code>--fastscore</code> command from the PRSice script. Run the following script to perform high-resolution scoring, and check the output files. </p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>Rscript ./Software/PRSice.R \
     --prsice Software/PRSice_mac \
     --base  Base_Data/GIANT_Height.txt \
     --target Target_Data/TAR \
     --snp MarkerName \
     --A1 Allele1 \
     --A2 Allele2 \
     --stat b \
     --beta \
     --pvalue p \
     --pheno Target_Data/TAR.height \
     --binary-target F \
     --perm 1000 \
     --out Results/Height.highres
</code></pre></div>
<p>When PRSice performs high-resolution scoring, it will generate a plot (Fig.1.2) presenting the model fit of PRS calculated at a large number of <em>P</em>-value thresholds.</p>
<p><img alt="High Resolution plot generated by PRSice" src="../../images/PRSice_HIGH-RES_PLOT.png" /></p>
<div class="admonition questions">
<p class="admonition-title">Questions</p>
<p>Which is the most predictive threshold?</p>
<p>How much better is the threshold identified using high-resolution scoring, in terms of model <span class="arithmatex">\(R^2\)</span>?</p>
<p>What is the empirical P-value and is it significant? </p>
<p>Should we use the raw P-value or the empirical <span class="arithmatex">\(P\)</span>-value? Why? </p>
<p>Which command line option generated the empirical <span class="arithmatex">\(P\)</span>-value? </p>
</div>
<div class="admonition information">
<p class="admonition-title">Information</p>
<p>To save time, we will not generate the empirical <span class="arithmatex">\(P\)</span>-value for most of the rest of this practical. However, when using PRSice for your analyses you should always generate the empirical <span class="arithmatex">\(P\)</span>-value to check whether the PRS association with the target phenotype is significant when multiple testing (across multiple thresholds) is accounted for.</p>
</div>
<h2 id="18-accounting-for-covariates">1.8 Accounting for Covariates</h2>
<p>When performing PRS analyses, we often want to account for the effect of covariates. 
Based on user inputs, PRSice can automatically incorporate covariates into its model. For example, the following commands will include sex into the regression model as a covariate:</p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>Rscript ./Software/PRSice.R \
     --prsice Software/PRSice_mac \
     --base  Base_Data/GIANT_Height.txt \
     --target Target_Data/TAR \
     --snp MarkerName \
     --A1 Allele1 \
     --A2 Allele2 \
     --stat b \
     --beta \
     --pvalue p \
     --pheno Target_Data/TAR.height \
     --binary-target F \
     --cov Target_Data/TAR.covariate \
     --cov-col Sex \
     --out Results/Height.sex
</code></pre></div>
<p>When covariates are included in the analysis, PRSice will use the model fit of <strong>only PRS</strong> for all its output. 
This <em>PRS.<span class="arithmatex">\(R^2\)</span></em> is calculated from a formula that provides an unbiased estimate of the difference between the <span class="arithmatex">\(R^2\)</span> of the null model (e.g. <span class="arithmatex">\(Height\sim Sex\)</span>) and the <span class="arithmatex">\(R^2\)</span> of the full model (e.g. <span class="arithmatex">\(Height\sim Sex+PRS\)</span>). </p>
<div class="admonition information">
<p class="admonition-title">Information</p>
<p>Usually, Principal Components (PCs) are also included as covariates in the analysis to account for population structure. However, it can be tedious to type all 20 or 40 PCs (e.g. PC1,PC2,...,PC20). 
In PRSice, you can add  <strong>@</strong> in front of the<code>--cov-col</code> string to activate the automatic substituation of numbers. 
If <strong>@</strong> is found in front of the  <code>--cov-col</code> string, then any numbers within <strong>[</strong> and <strong>]</strong> will be parsed. E.g. <code>@PC[1-3]</code> will be read as <em>PC1,PC2,PC3</em>. Discontinuous input are also supported: <code>@cov[1.3-5]</code> will be parsed as <em>cov1,cov3,cov4,cov5</em>. 
You can also mix it up E.g. <code>@PC[1-3],Sex</code> will be interpret as <em>PC1,PC2,PC3,Sex</em> by PRSice.</p>
</div>
<div class="admonition questions">
<p class="admonition-title">Questions</p>
<p>How does the inclusion of sex as a covariate change the results?</p>
</div>
<p>If you want to control for the effect of a <strong>categorical variable</strong>, such as socio-economic status (SES) or smoking status, then you can use the option <code>--cov-factor</code> in PRSice to automatically generate "dummy variables" corresponding to different levels of the factor. The following command shows how this can be done (using sex as an example):</p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>Rscript.exe ./Software/PRSice.R  \
     --prsice Software/PRSice_mac \
     --base  Base_Data/GIANT_Height.txt \
     --target Target_Data/TAR \
     --snp MarkerName \
     --A1 Allele1 \
     --A2 Allele2 \
     --stat b \
     --beta \
     --pvalue p \
     --pheno Target_Data/TAR.height \
     --binary-target F \
     --cov Target_Data/TAR.covariate \
     --cov-col Sex \
     --cov-factor Sex \
     --out Results/Height.sex
</code></pre></div>
<h2 id="19-case-control-studies">1.9 Case-Control Studies</h2>
<p>In the previous exercises, we performed PRS analyses on height, which is a quantitative trait. 
For binary phenotypes (e.g case-control) there are a number of differences in the analysis:</p>
<ol>
<li>Logistic regression has to be performed instead of linear regression</li>
<li>ORs are usually provided and should be converted to <span class="arithmatex">\(\beta\)</span>'s when constructing PRS so that they can be simply added across the genome</li>
</ol>
<p>Here we will use CAD as an example.
You will find the summary statistics under <code>Base_Data</code> (<strong>Cardio_CAD.txt</strong>) and the phenotype file (<strong>TAR.cad</strong>) under <code>Target_Data</code>.
You will also need to specify <code>--binary-target T</code> in the PRSice command to indicate that the phenotype is binary. </p>
<div class="admonition information">
<p class="admonition-title">Information</p>
<p>GWAS summary statistics for binary traits tend to report the OR instead of the <span class="arithmatex">\(\beta\)</span> coefficient, in which case the <code>--or</code> should be used.
However, CARDIoGRAM plus C4D consortium provided the <span class="arithmatex">\(\beta\)</span> coefficient, therefore we will include <code>--beta</code> in our code and specify <code>--binary-target T</code> to indicate that the phenotype is binary. </p>
</div>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>Rscript ./Software/PRSice.R \
     --prsice Software/PRSice_mac \
     --base  Base_Data/Cardio_CAD.txt \
     --target Target_Data/TAR \
     --snp markername \
     --A1 effect_allele \
     --A2 noneffect_allele \
     --chr chr \
     --bp bp_hg19 \
     --stat beta \
     --beta \
     --pvalue p_dgc \
     --binary-target T \
     --pheno Target_Data/TAR.cad \
     --perm 100 \
     --out Results/CAD.highres
</code></pre></div>
<div class="admonition information">
<p class="admonition-title">Information</p>
<p>If <code>--chr</code> and <code>--bp</code> are added to the command line (as above) then PRSice can check whether the SNPs in the Base and Target data have the same chromosomal coordinates. </p>
</div>
<div class="admonition questions">
<p class="admonition-title">Questions</p>
<p>What is the <span class="arithmatex">\(R^2\)</span> and <span class="arithmatex">\(P\)</span>-value of the best-fit PRS?</p>
<p>Does this suggest that there is a significant association between the CAD PRS and CAD status in the target sample? </p>
</div>
<h2 id="110-generating-prs-quantile-plots">1.10 Generating PRS Quantile Plots</h2>
<p>When considering the potential clinical utility of PRS, researchers often want to see the risk of disease among individuals in e.g. the top 5% or top 1% of PRS values compared to the rest of the population. Quantile plots are a nice way to illustrate the potential clinical utility of PRS (Fig.1.3) - made popular in the landmark PRS paper on the effects of PRS on cardiovascular disease by Khera et al (2018) - but should also be interpreted with some caution (see Fig.5 in: Choi, Mak &amp; O'Reilly 2020).</p>
<p>To generate quantile plots in PRSice, simply add the <code>--quantile</code> option and specify the number of quantiles that you would like to plot (e.g.  <code>--quantile 10</code> for deciles).</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you have already calculated the PRS using a previous command, then we can skip repeating the same calculation by using the <code>--plot</code> option</p>
</div>
<p><img alt="Quantile plot generated by PRSice" src="../../images/PRSice_QUANTILES_PLOT.png" /></p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>Rscript ./Software/PRSice.R \
    --prsice Software/PRSice_mac \
    --base Base_Data/Cardio_CAD.txt \
    --target Target_Data/TAR \
    --snp markername \
    --A1 effect_allele \
    --A2 noneffect_allele \
    --chr chr \
    --bp bp_hg19 \
    --stat beta \
    --beta \
    --pvalue p_dgc \
    --binary-target T \
    --pheno Target_Data/TAR.cad \
    --perm 100 \
    --quantile 10 \
    --plot \
    --out Results/CAD.highres
</code></pre></div>
<div class="admonition information">
<p class="admonition-title">Information</p>
<p>The <code>--plot</code> option tells PRSice to generate the plots without re-running the whole PRSice analysis. 
This is useful when you want to change some of the parameters for plotting e.g. the number of quantiles. <strong>Try running the previous command with 20 quantiles - and again with 50 quantiles - checking the quantile plot each time.</strong></p>
</div>
<div class="admonition questions">
<p class="admonition-title">Questions</p>
<p>What is the approximate higher risk of CAD for individuals in the top 5% of CAD PRS compared to those individuals with an average CAD PRS?</p>
</div>
<p>Quantile plots only separate samples into quantiles of equal size. However, we can also use PRSice to produce a more flexible "strata plot", where quantiles of different sizes can be displayed in the same plot (e.g. Fig.1.3). We use  <code>--quant-break</code>, corresponding to the upper bound of each strata, and <code>--quant-ref</code>, which represents the upper bound of the reference quantile (best for interpretation of results if this is the central quantile):</p>
<p><img alt="Strata plot generated by PRSice" src="../../images/PRSice_STRATA_PLOT.png" /></p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>Rscript ./Software/PRSice.R \
    --prsice Software/PRSice_mac \
    --base Base_Data/Cardio_CAD.txt \
    --target Target_Data/TAR \
    --snp markername \
    --A1 effect_allele \
    --A2 noneffect_allele \
    --chr chr  \
    --bp bp_hg19 \
    --stat beta \
    --beta \
    --pvalue p_dgc \
    --binary-target T \
    --pheno Target_Data/TAR.cad \
    --perm 100 \
    --quantile 100 \
    --quant-break 1,5,10,20,40,60,80,90,95,99,100 \
    --quant-ref 60 \
    --plot \
    --out Results/CAD.highres
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can see these results as plots in the files *_QUANTILES_PLOT_* or 
*_STRATA_PLOT_*, while the quantile results can also be seen in Table form in the *_QUANTILES_* file.
The results shown in the quantile and strata plots here do not show much higher risk in the top quantiles, but this is because our simulated target data are small.</p>
</div>
<h2 id="111-cross-trait-analysis">1.11 Cross-Trait Analysis</h2>
<p>A popular application of PRS is in performing cross-trait analyses. 
This allows some interesting analyses such as those performed by Ruderfer et al 2014 (Fig.1.5), which used the bipolar PRS to predict into different clinical dimensions of schizophrenia.</p>
<p><img alt="Plot taken from Ruderfer et al. 2014" src="../../images/sczbp.png" /></p>
<p>In this practical, we will perform cross-trait analyses between CAD and Height, using height as the base and CAD as the target. </p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>Rscript ./Software/PRSice.R \
     --prsice Software/PRSice_mac \
     --base Base_Data/GIANT_Height.txt \
     --target Target_Data/TAR \
     --snp MarkerName \
     --A1 Allele1 \
     --A2 Allele2 \
     --stat b \
     --beta \
     --pvalue p \
     --pheno Target_Data/TAR.cad \
     --binary-target T \
     --out Results/Cross.highres
</code></pre></div>
<div class="admonition questions">
<p class="admonition-title">Questions</p>
<p>What is the <span class="arithmatex">\(R^2\)</span> for the most predictive threshold when using height as the base phenotype and CAD as the target phenotype?</p>
<p>Now try using CAD as the base to predict height as the target trait? What is the PRS <span class="arithmatex">\(R^2\)</span> for that? </p>
<p>Can you explain why the results might be different when the phenotypes have been switched between the base and target? Or do you think the results should be the same?</p>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>To perform a cross-trait analysis involving a large number of target traits, you can use the <code>--pheno-col</code> option, which allows multiple target phenotypes to be included in the analysis simultaneously.</p>
</div>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../../tut_plink/" class="btn btn-neutral float-left" title="Plink Tutorial"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../workshop_practical_tade/" class="btn btn-neutral float-right" title="Day1 (PM)">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/tadesouaiaia/prsWorkshop-website" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../../tut_plink/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../workshop_practical_tade/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="../../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "../..";</script>
    <script src="../../js/theme_extra.js"></script>
    <script src="../../js/theme.js"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-MML-AM_CHTML"></script>
      <script src="../../javascripts/details.js"></script>
      <script src="../../search/main.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
